### ISSUE TITLE
.appimage is slow to start — include .pyc into the bundle

### ISSUE THREAD

**Issue by ValdikSS** [Electron-Cash/Electron-Cash#2903]
AppImage (Linux) version of electron-cash is rather slow to start, it takes about 3 seconds on my modern Intel 1340P machine.
This is because Python bytecode (`.pyc` files) are not included into .appimage, and since appimage could not be updated, Python compiles them every run.

If .appimage is exctracted with `./Electron-Cash-4.4.1-x86_64.AppImage --appimage-extract` and ran from the directory (so it could update itself), subsequent runs are vastly faster.

```
$ time ./AppRun daemon --help
…
real    0m1,626s
user    0m1,274s
sys     0m0,065s

$ time ./AppRun daemon --help
…
real    0m0,516s
user    0m0,287s
sys     0m0,052s
```

Please run the application once in your building process, and include all the built `.pyc` files into .appimage.`

**Comment by ValdikSS** [comment]
The other appimage-related optimization is to use zstd compression.
Current "stable" version of AppImageKit does not support it, so new appimagetool should be used
https://github.com/AppImage/appimagetool/

**Comment by cculianu** [comment]
Don't even need to run it.. I think there's a python cli arg to just compile everything...

**Comment by EchterAgo** [comment]
`squashfs-tools` and the kernel support `zstd` for SquashFS since 2017. `appimagetool` just passes through the chosen compression to `mksquashfs`.

I benchmarked all the available compression options including `zstd` a while ago and `gzip` was the fastest, IIRC `zstd` took quite a bit longer to start, but that was likely before https://github.com/AppImage/appimagetool/commit/81ee1db4946780292cd046d4c269a9b4fd7bc29d

As for keeping the bytecode files in the AppImage, we can just initiate a compilation for every `.py` file at build time. Note that the AppImage is already quite large and if this significantly increases size I'm sure some people will complain. IIRC it might be enough to keep just the `.pyc` files, which might even decrease size.

**Comment by ValdikSS** [comment]
Yes, seems there's [compileall](https://docs.python.org/3/library/compileall.html) module.
`python -m compileall .`, something like that.
